test/unit/cmap/commands.test.js

```diff
@@ -0,0 +1,92 @@
+const { expect } = require('chai');
+const { Response } = require('../../../src/cmap/commands');
+
+describe('commands', function () {
+  describe('Response', function () {
+    describe('#parse', function () {
+      context('when the message body is invalid', function () {
+        const message = Buffer.from([]);
+        const header = {
+          length: 0,
+          requestId: 0,
+          responseTo: 0,
+          opCode: 0
+        };
+        const body = Buffer.from([]);
+
+        it('throws an exception', function () {
+          const response = new Response(message, header, body);
+          expect(() => response.parse()).to.throw();
+        });
+      });
+    });
+
+    describe('#constructor', function () {
+      context('when the message body is invalid', function () {
+        const message = Buffer.from([]);
+        const header = {
+          length: 0,
+          requestId: 0,
+          responseTo: 0,
+          opCode: 0
+        };
+        const body = Buffer.from([]);
+
+        it('does not throw an exception', function () {
+          let error;
+          try {
+            new Response(message, header, body);
+          } catch (err) {
+            error = err;
+          }
+          expect(error).to.be.undefined;
+        });
```

Comment:
Let's add a test case that reproduced the original bug:
```js
const b = Buffer.alloc(5 * 4) // 5 int32s
b.writeInt32LE(-1, 16) // numReturned location
/// constructor doesn't throw, parse does with RangeError
```

Reply:
Test case added and checks error type and message.